amplipi/streams.py

```diff
@@ -973,6 +973,72 @@ def info(self) -> models.SourceInfo:
     source.track = "Not currently supported"
     return source
 
+class Aux(BaseStream):
+  """ A stream to play from the aux input. """
+  def __init__(self, name: str, disabled: bool = False, mock: bool = False):
+    super().__init__('aux', name, disabled=disabled, mock=mock)
+    self.bkg_thread = None
+
+  def reconfig(self, **kwargs):
+    reconnect_needed = False
+    if 'disabled' in kwargs:
+      self.disabled = kwargs['disabled']
+    if 'name' in kwargs:
+      self.name = kwargs['name']
+    if reconnect_needed:
+      last_src = self.src
+      self.disconnect()
+      time.sleep(0.1) # delay a bit, is this needed?
+      self.connect(last_src)
+
+  def connect(self, src):
+    """ Connect a short run VLC process with audio output to audio source """
+    print(f'connecting {self.name} to {src}...')
+
+    if self.mock:
+      self._connect(src)
+      # make a thread that waits for a couple of secends and returns after setting info to stopped
+      self.bkg_thread = threading.Thread(target=self.wait_on_proc)
+      self.bkg_thread.start()
+      return
+
+    # Set input source
+    amixer_command = f'amixer -D usb71 set "PCM Capture Source" "Line"'
+
+    subprocess.check_call(amixer_command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+    # Start audio via runvlc.py
+    vlc_args = f'cvlc -A alsa --alsa-audio-device {utils.real_output_device(src)} alsa://plughw:cmedia8chint,0 vlc://quit'
+    print(f'running: {vlc_args}')
+    self.proc = subprocess.Popen(args=vlc_args.split(), stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    self._connect(src)
+    # make a thread that waits for a couple of secends and returns after setting info to stopped
+    self.bkg_thread = threading.Thread(target=self.wait_on_proc)
+    self.bkg_thread.start()
+    return
+
+  def wait_on_proc(self):
+    """ Wait for the vlc process to finish """
+    if self.proc is not None:
+      self.proc.wait() # TODO: add a time here
+    else:
+      time.sleep(0.3) # handles mock case
+    self.state = 'stopped' # notify that the audio is done playing
```

Comment:
wait on proc is used for announcements, I'm not sure how useful it is here. Realistically we don't need it but it may be useful for debugging whether or not the aux is still playing.